services/brig/src/Brig/API/Federation.hs

```diff
@@ -118,11 +120,14 @@ sendConnectionAction originDomain NewConnectionRequest {..} = do
     else pure NewConnectionResponseUserNotActivated
 
 getUserByHandle ::
-  Member GalleyProvider r =>
+  ( Member GalleyProvider r,
+    Member FederationConfigStore r
+  ) =>
   Domain ->
   Handle ->
   ExceptT Error (AppT r) (Maybe UserProfile)
 getUserByHandle domain handle = do
+  -- TODO: should we do the team federation check here, too?
```

Comment:
Confirm with someone from the product team, but I think the check should be performed here.

Reply:
I'll remove this comment and create a ticket instead. This should be a separate PR, IMO.

Reply:
After looking into this, I think, that it is not necessary to apply the team federation policy here. The result of the design was that we needed to prevent connections and search in case of a team federation that is not allowed. The `getUserByHandle` is part of a federated handler and cannot be called by clients directly anyway. So this should be fine. Let me know if you don't agree.